src/main/java/internetshop/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -29,7 +28,12 @@ public ShoppingCart addProduct(ShoppingCart shoppingCart, Product product) {
 
     @Override
     public boolean deleteProduct(ShoppingCart shoppingCart, Product product) {
-        return shoppingCartDao.get(shoppingCart.getId()).get().getProducts().remove(product);
+        if (shoppingCart.getProducts().removeIf(s -> s.getId().equals(product.getId()))) {
```

Comment:
try to add the same product to your cart a few times and then delete it

Reply:
oh, really))) thank you) I really appreciate that you approved my hw with this mistake!